_zh-cn/tour/higher-order-functions.md

```diff
@@ -13,3 +13,98 @@ language: zh-cn
 next-page: nested-functions
 previous-page: mixin-class-composition
 ---
+
+高阶函数是指使用其他函数作为参数、或者返回一个函数作为结果的函数。在Scala中函数是一等类型值，所以允许定义高阶函数。
+这里的术语可能有点让人困惑：使用函数作为参数，或者返回值为函数的“函数”和“方法”，我们均称之为“高阶函数”。
+
+一个最简单的例子是Scala中对集合的高阶函数`map`
+```
+val salaries = Seq(20000, 70000, 40000)
+val doubleSalary = (x: Int) => x * 2
+val newSalaries = salaries.map(doubleSalary) // List(40000, 140000, 80000)
+```
+函数`doubleSalary`有一个整型参数`x`，返回`x * 2`。一般来说，在`=>`左边的元组是函数的参数列表，而在右边的表达式是函数的返回值。
+在第3行，函数`doubleSalary`被应用在数组`salaries`中的每一个元素。
+
+为了简化压缩代码，我们可以使用匿名函数，直接作为参数传递给`map`:
+```
+val salaries = Seq(20000, 70000, 40000)
+val newSalaries = salaries.map(x => x * 2) // List(40000, 140000, 80000)
+```
+注意在上述示例中`x`没有被显式声明为Int类型，这是因为编译器能够根据map函数期望的类型推断出`x`的类型。对于上述代码，一种更惯用的写法为：
+```tut
+val salaries = Seq(20000, 70000, 40000)
+val newSalaries = salaries.map(_ * 2)
+```
+既然Scala编译器已经知道了参数的类型（一个单独的Int），你可以只给出函数的右半部分，不过需要使用`_`来替代唯一的参数名（在上一个例子中是`x`）
+
+## 强制转换方法为函数
+你同样可以传入一个对象方法作为高阶函数的参数，这是因为Scala编译器会将方法强制转换为一个函数。
+```
+case class WeeklyWeatherForecast(temperatures: Seq[Double]) {
+
+  private def convertCtoF(temp: Double) = temp * 1.8 + 32
+
+  def forecastInFahrenheit: Seq[Double] = temperatures.map(convertCtoF) // <-- passing the method convertCtoF
+}
+```
+在这个例子中，方法`convertCtoF`被传入`forecastInFahrenheit`。
+这是可以的，因为编译器强制将方法`convertCtoF`转成了函数`x => convertCtoF(x)` 
+（注: `x`是编译器生成的变量名，保证在其作用域是唯一的）。
+
+## 接收函数作为参数的函数
+使用高阶函数的一个原因是减少冗余的代码。比如说你想要写一个函数，提高一个人的工资。如果没有高阶函数，代码可能像这样：
```

Comment:
```
-比如说你想要写一个函数，提高一个人的工资。如果没有高阶函数
+比方说需要写几个方法以通过不同方式来提升员工工资，若不使用高阶函数
```

Reply:
Great, your translation is more in line with Chinese language habits. Thanks!